view/frontend/web/js/product/button.js

```diff
@@ -0,0 +1,52 @@
+define([
+    'jquery',
+    'underscore'
+],function ($, _) {
+
+    return function (config) {
+
+        $(config.buttonId).on('mouseover',function (e) {
+
+            var radioAndCheckbox = {};
+            var inputElements = [];
+
+            $.each($('.product-options-wrapper [aria-required="true"]'),function () {
+
+                if(typeof $(this).prop("name") === "undefined" || $(this).attr("type") === "hidden"){
+                    return;
+                }
+
+                if($(this).attr("type") === "radio" || $(this).attr("type") === "checkbox"){
+
+                    String($(this).attr("name")) in radioAndCheckbox ?
+                        radioAndCheckbox[$(this).attr("name")].push($(this).is(':checked'))
+                        : radioAndCheckbox[$(this).attr("name")] = [$(this).is(':checked')]
+                    ;
+
+                } else {
+
+                    inputElements.push(Boolean($(this).val()));
+                }
+            });
+
+            inputElements = _.every(inputElements);
+
+            $.each(radioAndCheckbox,function (i,el) {
```

Comment:
what do i and el refer to? could you make the names more descriptive? Also the name 'radioAndCheckbox' doesn't really say much about what it is either, radioAndCheckboxElements would make it more consistent with inputElements

Reply:
Yes, that's bad naming, **i** refers to element name, and **el** to value set, which is array and to be valid needs to contain one member value of boolean true. Will change to **name** and **valueSet** is that ok?